### PR TITLE
1.20.x: kernel updates, statically linked modprobe

### PR DESCRIPTION
**Pull request by larvacea**
**Description of changes:**

* Update kernels to 6.1.90, 5.15.158, and 5.10.216 ([#3976], [#3972])
* Include statically-linked packages/kmod modprobe ([#3981])

**Testing done:**

Kernel changes: our standard sonobuoy tests and manual inspection of patch and configuration changes.
modprobe: see [#3981] for details.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by yeazelm — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# v1.20.1
+
+## OS Changes
+* Update kernels to 6.1.90, 5.15.158, and 5.10.216 ([#3976], [#3972])
+* Include statically-linked packages/kmod ([#3981])
+
```
Comment (ginglis13) [inline CHANGELOG.md:10]:
```suggestion

[#3972]: https://github.com/bottlerocket-os/bottlerocket/pull/3972
[#3976]: https://github.com/bottlerocket-os/bottlerocket/pull/3976
[#3981]: https://github.com/bottlerocket-os/bottlerocket/pull/3981

```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+# v1.20.1
+
+## OS Changes
+* Update kernels to 6.1.90, 5.15.158, and 5.10.216 ([#3976], [#3972])
+* Include statically-linked packages/kmod ([#3981])
```
Comment (ginglis13) [inline CHANGELOG.md]:
```suggestion
* Include statically linked version of kmod ([#3981])
```

**Review by ginglis13 — CHANGES_REQUESTED** [review]

**Comment by larvacea** [comment]
I have incorporated @ginglis13's suggested changes.

**Review by ginglis13 — APPROVED** [review]

**Comment by hubertbits** [comment]
Would it be possible to publish a 1.20.1 release with just these changes so we can select this AMI in our launch template? Reading through the merged issues, cilium should be working again with these commits.

Thanks @vigh-m (and any unmentioned contributor) for investigating and fixing this issue!

**Comment by yeazelm** [comment]
Hello @obirhuppertz, we did release [1.20.2](https://github.com/bottlerocket-os/bottlerocket/releases/tag/v1.20.2) which provides a fix for CIlium. Please check that release and follow up on https://github.com/bottlerocket-os/bottlerocket/issues/3968 if you experience more issues with Cilium. Thanks!

**Comment by hubertbits** [comment]
@yeazelm @vigh-m can confirm cilium is still working on 1.20.2 bottlerocket nodes having the workaround via extraMounts and volumes removed in values.yaml. For me the issue is fixed now. Great work! :partying_face:

**Pull request merged**